xoinvader/animation.py

```diff
@@ -0,0 +1,101 @@
+"""Animation.
+
+Animation is set of keyframes.
+Value of selected attribute changes in time.
+
+Keyframe:
+  (time, value)
+
+Objects have animation manager which manages animation graph and switching."""
+
+
+from operator import itemgetter
+
+from xoinvader.utils import Timer
+
+
+# TODO: Implement animation graph and etc
+class AnimationManager(object):
+    """Manage list of object animation."""
+
+    def __init__(self):
+        self._animations = {}
+        self._current = None
+
+    def bind(self, name, obj):
+        """Transfer animation to other object."""
+        self._animations[name].bind(obj)
+
+    def add(self, name, *args, **kwargs):
+        """Add new animation, pass args to Animation class.
+
+        :param str name: animation name
+        """
+
+        self._animations[name] = Animation(name, *args, **kwargs)
+        if len(self._animations) == 1:
+            self._current = name
+
+    def __iter__(self):
+        return self
+
+    def __next__(self):
+        try:
+            next(self._animations[self._current])
+        except StopIteration:
+            raise  # We need to change animation
+
+
+class Animation(object):
+    """Animation unit.
+
+    :param str name: animation name
+    :param object bind: object to bind animation
+    :param str attr: attribute to change in frames
+    :param list keyframes: {float: object} dicts
+    :param bool loop: loop animation or not
+    """
+
+    def __init__(self, name, bind, attr, keyframes, loop=False):
+        self._name = name
+        self._obj = bind
+        self._attr = attr
+        self._keyframes = sorted(keyframes, key=itemgetter(0))
+        self._loop = loop
+        if not keyframes:
```

Comment:
...So it immediately rises.
I just don't get it, why we need to call `sorted` on maybe empty list before performing emptiness check.